lib/src/chart/line_chart/line_chart_painter.dart

```diff
@@ -166,19 +165,40 @@ class LineChartPainter extends AxisChartPainter<LineChartData>
   }
 
   void _drawBarLine(Canvas canvas, Size viewSize, LineChartBarData barData) {
-    final barPath = _generateBarPath(viewSize, barData);
-
-    final belowBarPath = _generateBelowBarPath(viewSize, barData, barPath);
-    final completelyFillBelowBarPath =
-        _generateBelowBarPath(viewSize, barData, barPath, fillCompletely: true);
+    final List<List<FlSpot>> barList = [[]];
 
-    final aboveBarPath = _generateAboveBarPath(viewSize, barData, barPath);
-    final completelyFillAboveBarPath =
-        _generateAboveBarPath(viewSize, barData, barPath, fillCompletely: true);
+    // handle nullability by splitting off the list into multiple
+    // separate lists when separated by nulls
+    // and ignore nulls when they're first or last
+    for (int i = 0; i < barData.spots.length; i++) {
+      if (barData.spots[i].x != null && barData.spots[i].y != null) {
+        barList.last.add(barData.spots[i]);
+      } else {
+        if (i != 0 && i != barData.spots.length - 1) {
+          barList.add(<FlSpot>[]);
+        }
+      }
+    }
 
-    _drawBelowBar(canvas, viewSize, belowBarPath, completelyFillAboveBarPath, barData);
-    _drawAboveBar(canvas, viewSize, aboveBarPath, completelyFillBelowBarPath, barData);
-    _drawBar(canvas, viewSize, barPath, barData);
+    // paint each sublist that was built above
+    // bar is passed in seperately from barData
+    // because barData is the whole line
+    // and bar is a piece of that line
+    barList.forEach((bar) {
+      final barPath = _generateBarPath(viewSize, barData, bar: bar);
+      _drawBar(canvas, viewSize, barPath, barData);
```

Comment:
Why you changed the draw order?
previously _drawBar was the last one.

Reply:
If it is for readability, please make it required with `@required` [annotation](https://dart.dev/guides/language/language-tour#optional-parameters).
And I think `bar` is not descriptive enough, I think `barSpots` is more descriptive.
Thanks!

Reply:
Good catch

Reply:
Hmm 'bar' makes sense here because in the forEach it's iterating a barList so the two names connect. barSpotsList would be out of context

Reply:
I'll change the variable name to barSpots, I think that's what you were asking for

Reply:
done